### PR TITLE
Register `read_parquet` and `read_csv` with dask-expr

### PR DESCRIPTION
**Pull request by rjzamora**
## Description
After https://github.com/dask/dask-expr/pull/1114, Dask cuDF must register specific `read_parquet` and `read_csv` functions to be used when query-planning is enabled (the default).

**This PR is required for CI to pass with dask>2024.8.0**

**NOTE**: It probably doesn't make sense to add specific tests for this change. Once the 2014.7.1 dask pin is removed, all `dask_cudf` tests using `read_parquet` and  `read_csv` will fail without this change...

## Checklist
- [ ] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.

### CODE REVIEW

File: python/dask_cudf/dask_cudf/backends.py [hunk 1]
```diff
@@ -667,6 +667,41 @@ def from_dict(
             constructor=constructor,
         )
 
+    @classmethod
```
Comment (madsbk) [inline python/dask_cudf/dask_cudf/backends.py:670]: nitpick: use `staticmethod` like in the other read methods?

```suggestion
    @staticmethod
```

**Review by madsbk — APPROVED** [review]
Looks good, only have a minor suggestion

File: python/dask_cudf/dask_cudf/backends.py [hunk 2]
```diff
@@ -667,6 +667,41 @@ def from_dict(
...
+    def read_parquet(cls, *args, engine=None, **kwargs):
```
Comment (rjzamora) [inline python/dask_cudf/dask_cudf/backends.py:671]:
```suggestion
    def read_parquet(*args, engine=None, **kwargs):
```

**Review by quasiben — APPROVED** [review]

**Pull request merged**